tests/lib/openstack_query/handlers/test_client_side_handler.py

```diff
@@ -0,0 +1,207 @@
+        self.assertTrue(
+            self.instance.check_supported(
+                MockQueryPresets.ITEM_1, MockProperties.PROP_1
+            )
+        )
+
+        # when preset has explicit prop mapping
+        self.assertTrue(
+            self.instance.check_supported(
+                MockQueryPresets.ITEM_2, MockProperties.PROP_2
+            )
+        )
+
+    def test_check_preset_supported_false(self):
+        # when preset is not supported
+        self.assertFalse(
+            self.instance.check_supported(
+                MockQueryPresets.ITEM_4, MockProperties.PROP_1
+            )
+        )
+
+        # when preset found, but prop is not supported
+        self.assertFalse(
+            self.instance.check_supported(
+                MockQueryPresets.ITEM_2, MockProperties.PROP_3
+            )
+        )
+
+    @patch(
+        "openstack_query.handlers.client_side_handler.ClientSideHandler._check_filter_func"
+    )
+    @patch(
+        "openstack_query.handlers.client_side_handler.ClientSideHandler._filter_func_wrapper"
+    )
+    def test_get_filter_func_valid(
+        self, mock_filter_func_wrapper, mock_check_filter_func
+    ):
+        # define inputs
+        mock_prop_func = MagicMock()
+        mock_kwargs = {"arg1": "val1", "arg2": "val2"}
+
+        # mock function return values
+        mock_check_filter_func.return_value = True, ""
+        mock_filter_func_wrapper.return_value = "a-boolean-value"
+
+        res = self.instance.get_filter_func(
+            MockQueryPresets.ITEM_1, MockProperties.PROP_1, mock_prop_func, mock_kwargs
+        )
+        val = res("test-openstack-res")
+
+        # MockQueryPresets.ITEM_1 and MockProperties.PROP_1 are valid and should return 'item1_func'
+        # - which represents a function mapping
+        mock_check_filter_func.assert_called_once_with("item1_func", mock_kwargs)
+        mock_filter_func_wrapper.assert_called_once_with(
+            "test-openstack-res", "item1_func", mock_prop_func, mock_kwargs
+        )
+
+        self.assertEqual(val, "a-boolean-value")
+
+    @patch(
+        "openstack_query.handlers.client_side_handler.ClientSideHandler._check_filter_func"
+    )
+    def test_get_filter_func_invalid(self, mock_check_filter_func):
+        mock_prop_func = MagicMock()
+        mock_kwargs = {"arg1": "val1", "arg2": "val2"}
+
+        # when preset is invalid
+        with self.assertRaises(QueryPresetMappingError):
+            self.instance.get_filter_func(
+                MockQueryPresets.ITEM_4,
+                MockProperties.PROP_1,
+                mock_prop_func,
+                mock_kwargs,
+            )
+
+        # when the preset is valid, but property is invalid
+        with self.assertRaises(QueryPresetMappingError):
+            self.instance.get_filter_func(
+                MockQueryPresets.ITEM_2,
+                MockProperties.PROP_3,
+                mock_prop_func,
+                mock_kwargs,
+            )
```

Comment:
Can we split this into separate test cases, so it's obvious if something breaks it will either be
`test_get_filter_func_invalid_prop`
`test_get_filter_func_invalid_preset`